pandasaurus_cxg/anndata_analyzer.py

```diff
@@ -0,0 +1,50 @@
+from enum import Enum
+
+import pandas as pd
+
+from pandasaurus_cxg.anndata_loader import AnndataLoader
+from pandasaurus_cxg.schema.schema_loader import read_json_file
+
+
+class AnndataAnalyzer:
+    def __init__(self, file_path: str, schema_path: str):
+        self._anndata_obs = AnndataLoader.load_from_file(file_path).obs
+        self._schema = read_json_file(schema_path)
+
+    def co_annotation_report(self):
+        free_text_cell_type = [key for key, value in self._schema.items() if value]
+        temp_result = []
+        for text in free_text_cell_type:
+            if text in self._anndata_obs.columns:
+                co_oc = (
+                    self._anndata_obs[[text, "cell_type"]].drop_duplicates().reset_index(drop=True)
+                )
+                predicate_dict = co_oc.groupby("cell_type")[text].apply(list).to_dict()
+                co_oc["predicate"] = co_oc.apply(
+                    lambda row: Predicate.CLUSTER_MATCHES.value
+                    if row[text] in predicate_dict.get(row["cell_type"], [])
+                    and len(predicate_dict.get(row["cell_type"], [])) == 1
+                    else (
```

Comment:
Bit opaque. I think this works, but looks like this only tests co-annotation with cell_type field. Should be looking at all fields.

Reply:
All fields as in tissue, diseases and organism etc. or just all other free text cell type field?

Reply:
Just cell type fields

Reply:
All combinations of pairs free text and ontology cell type fields.